### DISCUSSION TITLE
AD method and the rapid development of a new adjoint solver

### DISCUSSION THREAD

**Discussion by chairmanmao256** [mdolab/dafoam#182] [General]
Dear He Ping, I've just read your article: _DAFoam: An Open-Source Adjoint Framework for Multidisciplinary Design Optimization with OpenFOAM_. It is mentioned in the article that DAFoam implements a solver-agnostic FD method to compute partial derivatives in the adjoint equation. This feature enables developers of DAFoam to create new PDE solver together with its adjoint rapidly. 
However, in v2.2.0 and newer versions of DAFoam, AD is implemented to compute the partial derivatives. Does the AD method compromise the convenience of adding a new solver that is described in your article?(or, can I use the method described in your article mentioned earlier to add a new solver and its adjoint successfully?)
Thank you!!

**Comment by friedenhe** [comment]
No, the AD does not compromise the quick-adjoint implementation capability in DAFoam. You can still follow a similar route to implement a new adjoint solver.

**Reply by chairmanmao256** [reply]
Got it! Thank you!
